tests/e2e/APNSNetworkServiceTest.php

```diff
@@ -44,6 +44,39 @@ public function testThatSendingAfterGoAwayFrameEmitsIsRecoverable() {
 		$service->close_connection();
 	}
 
+	public function testThatServerTimeoutIsRecoverable() {
+
+		$service = ( new APNSNetworkService() )
+			->set_certificate_bundle_path( dirname( __DIR__ ) . '/MockAPNSServer/test-cert.pem' )
+			->set_port( 8443 )
+			->set_timeout( 1 );
+
+		$service->enqueue_request( 'https://127.0.0.1/', [], '' );
+		// It's easier to test a remote server here rather than make node work how we want
+		$service->enqueue_request( 'https://httpbin.org/delay/10', [], '' );
```

Comment:
Fair enough. And since we're testing the timeout behavior, if the network is down or there's any other of those annoying issues when hitting a remote service, we'll still trigger the failure behavior 😅 

Besides, this is an E2E test, so it's kinda okay to hit a remote server.

Reply:
Yeah, I'd like this to work with the mock server, but any timeout stuff I do causes the whole batch to fail, which seems incorrect? Though it's really hard to know what the reference behaviour should be here